lib/node_modules/@stdlib/math/base/special/normhermite-polynomial/lib/main.js

```diff
@@ -18,20 +18,63 @@
 
 'use strict';
 
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+
+
 // MAIN //
 
 /**
 * Evaluates a normalized Hermite polynomial.
 *
-* @param {<TODO: type>} TODO - TODO: param desc
-* @returns {<TODO: type>} TODO: desc
+* @param {integer} n - degree of polynomial
+* @param {number} x - evaluation point
+* @returns {number} function value
+*
+* @example
+* var v = normhermitepoly( 1, 0.5 );
+* // returns 0.5
+*
+* @example
+* var v = normhermitepoly( -1, 0.5 );
+* // returns 0.0
 *
 * @example
-* var v = normhermitepoly( 1, 1.0 );
-* // returns TODO
+* var v = normhermitepoly( 0, 0.5 );
+* // returns 1.0
+*
+* @example
+* var v = normhermitepoly( 2, 0.5 );
+* // returns -0.75
 */
-function normhermitepoly() {
-	// TODO: implementation
+function normhermitepoly( n, x ) {
+	var y1;
+	var y2;
+	var y3;
+	var i;
+
+	if ( isnan( n ) || isnan( x ) ) {
+		return NaN;
+	}
+	if ( n === 0 ) {
+		// The x gets completely cancelled from the expression
+		return 1.0;
+	}
+	if ( n < 0 ) {
```

Comment:
@jrreed83 This doesn't seem right to me. If this is adapted from the reference implementation, then my best guess would be that they're intercepting even/oddness at a wrapper level so that the low-level implementation is never actually called with a negative number. In that sense, this would maybe be better thought of as an impossible state. That's just a guess since I'm not able to find any rationalization for why this would be zero. Trying out [scipy.special.eval_hermitenorm](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.eval_hermitenorm.html#scipy.special.eval_hermitenorm), I get:

```
>>> eval_hermitenorm(0, -2.0)
1.0
>>> eval_hermitenorm(1, -2.0)
-2.0
>>> eval_hermitenorm(2, -2.0)
2.9999999999999991
```

as I think would be expected. There's probably a better way, it seems like a simple way to handle negative numbers could be (cleaning it up a bit of course, maybe with [is-even](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/assert/is-even) unless that's overkill),

```javascript
var out = (x*y2) - y3;
return n % 2 === out : -out;
```

Reply:
@rreusser I believe @jrreed83 is talking about whether `n`, the polynomial degree, should be allowed to be negative.